File: UniStay.Api/Models/SavedRoom.cs
using System.ComponentModel.DataAnnotations;

namespace UniStay.Api.Models
{
    public class SavedRoom
    {
        [Key]
        public int SaveId { get; set; }
        public string UserId { get; set; } = string.Empty;
        public string RoomId { get; set; } = string.Empty;
        public long SavedAt { get; set; } = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
    }
}

File: app/src/main/java/com/unistay/android/domain/model/model/SavedRoom.kt
package com.unistay.android.domain.model.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "saved_rooms")
data class SavedRoom(
    @PrimaryKey(autoGenerate = true) val saveId: Int = 0,
    val userId: String,
    val roomId: String,
    val savedAt: Long = System.currentTimeMillis()
)